extension-api/extension-api.ts

```diff
@@ -363,8 +426,18 @@ export class DeployContext implements IDeployContext {
         this.scripts = [];
     }
 
-    public addEnvironmentVariables(vars: object) {
-        Object.assign(this.environmentVariables, vars);
+    public addEnvironmentVariables(envVars: EnvironmentVariables) {
+        const formattedVars = Object.keys(envVars)
+            .reduce((obj: any, name) => {
+                const envName = this.getInjectedEnvVarName(name);
+                obj[envName] = envVars[name];
+                return obj;
+            }, {});
+        Object.assign(this.environmentVariables, formattedVars);
+    }
+
+    public getInjectedEnvVarName(suffix: string) {
+        return `${this.serviceName}_${suffix}`.toUpperCase().replace(/-/g, '_');
```

Comment:
Yeah I believe we only allow alphanumeric plus '-' in names in Handel, thanks to the many, many AWS resource naming restrictions across services, so I think this works for me personally rather than pulling in constant-case.